cloudinit/config/schemas/schema-cloud-config-v1.json

```diff
@@ -3386,6 +3386,26 @@
                 "default": "''",
                 "description": "Optional content to write to the provided ``path``. When content is present and encoding is not 'text/plain', decode the content prior to writing. Default: ``''``"
               },
+              "source": {
+                "type": "object",
+                "description": "Optional specification for content loading from an arbitrary URI",
+                "additionalProperties": false,
+                "properties": {
+                  "uri": {
+                    "type": "string",
+                    "format": "uri",
+                    "description": "URI from which to load file content. If loading fails repeatedly, ``content`` is used instead."
+                  },
+                  "headers": {
```

Comment:
Cloud-init passes the value from headers to requests. Requests expects that the value should be string, bytestring, or unicode[1]. It would be ideal if we could validate that the inputs to cloud-init are valid types for requests.

If a user wants to pass an integer as a value and accidentally uses an unquoted integer (which in YAML will not be of type string), this change will benefit the user by validating the input and logging a warning about invalid configuration.

As it is, the header type isn't validated.

[1] https://requests.readthedocs.io/en/latest/user/quickstart/

Reply:
Aha, good point. Should we maybe just string-ify whatever we're passed? That might be a more forgiving option if the user forgets their quotes, but I could see it causing weirdness if they forget quotes and also supply, say, a value prefixed with `0o` for some reason (which would be decoded from octal, then string-ified out in decimal). It's a case of strictness versus weirdness, I suppose.

Reply:
Ah, okay, forcing string keys is much easier than I realized, and is probably the safest option anyway. Commit incoming...